share/functions/fish_command_not_found.fish

```diff
@@ -0,0 +1,61 @@
+### Command-not-found handlers
+# This can be overridden by defining a new fish_command_not_found function
+
+# Read the OS/Distro from /etc/os-release.
+# This has a "ID=" line that defines the exact distribution,
+# and an "ID_LIKE=" line that defines what it is derived from or otherwise like.
+# For our purposes, we use both.
+set -l os
+if test -r /etc/os-release
+    set os (string match -r '^ID(?:_LIKE)?\s*=.*' < /etc/os-release | \
+    string replace -r '^ID(?:_LIKE)?\s*=(.*)' '$1' | string trim -c '\'"' | string split " ")
+end
+
+# If an old handler already exists, defer to that.
+if functions -q __fish_command_not_found_handler
+    function fish_command_not_found
+        # The fish_command_not_found event was removed in fish 3.2.0,
+        # and future versions of fish will just call a function called "fish_command_not_found".
+        # You have defined a custom handler, we suggest renaming it to "fish_command_not_found".
+        __fish_command_not_found_handler $argv
+    end
+    # First check if we are on OpenSUSE since SUSE's handler has no options
+    # but the same name and path as Ubuntu's.
+else if contains -- suse $os || contains -- sles $os && type -q command-not-found
+    function fish_command_not_found
+        /usr/bin/command-not-found $argv[1]
```

Comment:
This path and the one from the `type -q`  above don't align. Using the basename should be fine, right? It's not in `/sbin` after all...

Reply:
Yeah, it's in $PATH, so we should use it via $PATH.